### DISCUSSION TITLE
Support xunit in FluxPoints.plot()

### DISCUSSION THREAD

**Discussion by mireianievas** [gammapy/gammapy#5219] [Help] [answered]
Using v1.0.1, but I believe it still applies to the development version.

Looking at the code (https://docs.gammapy.org/dev/_modules/gammapy/estimators/points/core.html#FluxPoints.plot ), it looks like like the units in the y-axis are hardcoded in the DEFAULT_UNIT dictionary, but I don't see a reference to the x-axis units. I believe it uses the default units stored in that FluxPoints object. 

It would be nice if, similar to how the SpectralModel.plot() work, it could take xunit and yunit as possible parameters and handle the conversions (e.g. TeV -> GeV).

**Comment by tunbehaun273** [comment]
Add: I would think it is also useful to be able to pass `flux_unit` as a kwarg to the plot() function again, as it was possible for v0.18.2. This could be done for the `FluxPoints` but also for the `SpectralModel`s in order to enable easily customizable plots.

**Comment by adonath** [comment, marked as answer]
@mireianievas Have you checked out the entry "Choose units for plotting" in our how to? (see https://docs.gammapy.org/1.1/user-guide/howto.html)

Removing the unit handling from the Gammapy API was a deliberate decision, because the combination of `matpplotlib` and `Astropy` already offers every functionality needed. Here is the copy and paste example  from the how to:

```python
import matplotlib.pyplot as plt
from gammapy.estimators import FluxPoints
from astropy import units as u

filename = "$GAMMAPY_DATA/hawc_crab/HAWC19_flux_points.fits"
fp = FluxPoints.read(filename)

ax = plt.subplot()
ax.xaxis.set_units(u.eV)
ax.yaxis.set_units(u.Unit("erg cm-2 s-1"))
fp.plot(ax=ax, sed_type="e2dnde")
```

The `ax.xaxis.set_units(u.eV)` and `ax.yaxis.set_units(u.eV)` pattern works with every(!) plot in Gammapy. Two things are important:

- You have to create an independent plot axis first using e.g. `plt.subplot()`
- The units have to be set before plotting, such that the plotted quantities can be converted to the right units
